src/utils/pathCheck.ts

```diff
@@ -0,0 +1,106 @@
+import fs from 'fs';
+import path from 'path';
+import childProcess from 'child_process';
+
+import semver from 'semver';
+import resources from '@/resources';
+
+const fsPromises = fs.promises;
+
+const flags: Record<string, string> = {
+  helm:    'version',
+  kim:     '-v',
+  kubectl: 'version',
+};
+const regexes: Record<string, RegExp> = {
+  // helm has to match both
+  // current: version.BuildInfo{Version:"v3.5.3", ...
+  // older:   Client: &version.Version{SemVer:"v2.16.12", ...
+  helm:    /Version.*:.*?"v(.+?)"/,
+  kim:     /version v(\S+)/,
+  kubectl: /Client Version.*?GitVersion:"v(.+?)"/,
+};
+
+export default async function shadowInfo(sourceDir: string, targetDir: string, binaryName: string): Promise<Array<string>> {
+  const notes: Array<string> = [];
+  // Don't have access to Electron.app in unit tests, so can't use the resources module
+  const referencePath = path.join(sourceDir, binaryName);
+
+  try {
+    await fsPromises.access(referencePath, fs.constants.R_OK | fs.constants.X_OK);
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      console.log(err);
+
+      return notes;
+    }
+  }
+  const proposedVersion = await getVersion(referencePath, binaryName);
+
+  if (!proposedVersion) {
+    return notes;
+  }
+  const paths: Array<string> = process.env.PATH?.split(path.delimiter) || [];
```

Comment:
```suggestion
  const paths: Array<string> = process.env.PATH?.split(path.delimiter) ?? [];
```
It's about the same thing, but shows clearer that we expect the value to be `string[]` or `undefined`.  (Also, to make sure you know that [`??`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) exists — it's relatively new.)

Reply:
sure, and I did know about `??`